source/digits_hits/include/GateRestrictedVolume.hh

```diff
@@ -0,0 +1,71 @@
+/** ----------------------
+  Copyright (C): OpenGATE Collaboration
+  This software is distributed under the terms
+  of the GNU Lesser General  Public Licence (LGPL)
+  See LICENSE.md for further details
+  ----------------------*/
+
+
+#ifndef GateRestrictedVolume_h
+#define GateRestrictedVolume_h 1
+
+#include "globals.hh"
+#include <iostream>
+
+#include "GateVPulseProcessor.hh"
+
+class GateRestrictedVolumeMessenger;
+
+
+/*! \class  GateRestrictedVolume
+    \brief  Pulse-processor modelling a simple volume discriminator.
+
+    - The method ProcessOnePulse of this class models a simple
+      volume discriminator: any input pulse whose volume name is not as
+      defined is copied into the output pulse-list.
+      On the contrary, any input pulse whose name is the same as 
+      specified is discarded.
+
+      \sa GateVPulseProcessor
+*/
+class GateRestrictedVolume : public GateVPulseProcessor
+{
+  public:
+
+    //! Constructs a new module attached to a GateDigitizer
+    GateRestrictedVolume(GatePulseProcessorChain* itsChain, const G4String& itsName);
+    //! Destructor
+    virtual ~GateRestrictedVolume();
```

Comment:
If you decide to use unique pointer (check my next comment) you can replace destructor with:
`virtual ~GateRestrictedVolume() = default;`

Reply:
I am wondering  what is the advantage of using unique pointers instead of default one? I am wondering if there is need to use them?

Reply:
The main advantage is that you don't care anymore about releasing the resources (deleting objects). It is handled automatically.  The undeleted pointers are one of the most common problems with memory leaks etc.

Reply:
This is a general remark. I haven't checked this particular case here.

Reply:
Unique pointer issue has been already solved